pkg/cloudprovider/vsphereparavirtual/route.go

```diff
@@ -0,0 +1,415 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vsphereparavirtual
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strings"
+	"time"
+
+	"sync"
+
+	"github.com/pkg/errors"
+
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	listerv1 "k8s.io/client-go/listers/core/v1"
+	clientretry "k8s.io/client-go/util/retry"
+
+	rest "k8s.io/client-go/rest"
+	cloudprovider "k8s.io/cloud-provider"
+	v1alpha1 "k8s.io/cloud-provider-vsphere/pkg/cloudprovider/vsphereparavirtual/apis/nsxnetworking/v1alpha1"
+	client "k8s.io/cloud-provider-vsphere/pkg/cloudprovider/vsphereparavirtual/client/clientset/versioned"
+	"k8s.io/cloud-provider-vsphere/pkg/util"
+	cloudnodeutil "k8s.io/cloud-provider/node/helpers"
+	klog "k8s.io/klog/v2"
+)
+
+// RoutesProvider is the interface definition for Routes functionality
+type RoutesProvider interface {
+	cloudprovider.Routes
+	AddNode(*v1.Node)
+	DeleteNode(*v1.Node)
+}
+
+type routesProvider struct {
+	routeClient client.Interface
+	namespace   string
+	nodeMap     map[string]*v1.Node
+	nodeMapLock sync.RWMutex
+	ownerRefs   []metav1.OwnerReference
+	kubeClient  clientset.Interface
+	nodeLister  listerv1.NodeLister
+}
+
+var _ RoutesProvider = &routesProvider{}
+
+const (
+	// AnnotationClusterName label is used to annotate GC name for Routes CR
+	AnnotationClusterName = v1alpha1.GroupName + "/clusterName"
+	// RealizedStateTimeout is the timeout duration for realized state check
```

Comment:
Remove these if not used

Reply:
https://github.com/kubernetes/cloud-provider-vsphere/pull/468/files#r643316448 I would need to uncomment those fields if we plan to merge the check for realized state with this PR